src/Models/NonhydrostaticModels/update_nonhydrostatic_model_state.jl

```diff
@@ -61,9 +62,17 @@ function compute_auxiliaries!(model::NonhydrostaticModel; p_parameters = tuple(p
     closure = model.closure
     diffusivity = model.diffusivity_fields
 
-    for (ppar, κpar) in zip(p_parameters, κ_parameters)
+    w_parameters = tuple(w_kernel_parameters(model.grid))
+
+    for (wpar, ppar, κpar) in zip(w_parameters, p_parameters, κ_parameters)
+        if !isnothing(model.free_surface)
+            compute_w_from_continuity!(model; parameters = wpar)
```

Comment:
I am unsure of the algorithm, but wouldn't this replace the w-velocity that should be prognostic?

Reply:
For sure. That's what is written in MITgcm docs. @jm-c can you confirm?